Add support for vJunosEvolved

akielaries opened the pull request:
Corresponding PR from [hellt/vrnetlab](https://github.com/hellt/vrnetlab/pull/152) for adding support for Juniper's vJunosEvolved, a virtual instance using PTX10001 as the reference. I used the below file to test and added as a lab example under `srlvjunos02`:

```yml
name: srlvjunos02

topology:
  nodes:
    srl:
      kind: nokia_srlinux
      image: ghcr.io/nokia/srlinux:23.7.1
      startup-config: srl.cli

    vevo:
      kind: vr-juniper_vjunosevolved
      image: vrnetlab/vr-vjunosevolved:23.2R1-S1.8-EVO
      startup-config: vjunos.cfg

  links:
    - endpoints: ["srl:e1-1", "vevo:eth1"]
    - endpoints: ["srl:e1-2", "vevo:eth2"]
    - endpoints: ["srl:e1-3", "vevo:eth3"]
```

hellt commented on README.md:
```diff
@@ -34,6 +34,7 @@ In addition to native containerized NOSes, containerlab can launch traditional v
 * [Juniper vMX](https://containerlab.dev/manual/kinds/vr-vmx/)
 * [Juniper vQFX](https://containerlab.dev/manual/kinds/vr-vqfx/)
 * [Juniper vSRX](https://containerlab.dev/manual/kinds/vr-vsrx/)
+* [Juniper vJunos-switch](https://containerlab.dev/manual/kinds/vr-vjunosswitch/)
```
@akielaries should you add here evo as well?

hellt commented on schemas/clab.schema.json:
```diff
@@ -62,6 +62,9 @@
                         "vr-vjunosswitch",
                         "vr-juniper_vjunosswitch",
                         "juniper_vjunosswitch",
+                        "vr-vjunosevolved",
+                        "vr-juniper_vjunosevolved",
```
safe to remove this. we gradually cleanup vr- prefixed names in favor of <vendor>_<nos> format

hellt reviewed (commented):
@akielaries thanks
You also have to add the line to `mkdocs.yml` file for the article to render in the list of kinds and lab examples

hellt commented:
you can also refresh the main branch and rebase your commits on top of it to resolve the testing issue with border0 failure.

akielaries commented:
Sure thing. I added the vJunosEvolved manual link to the README, a lab-example doc under [docs/lab-examples/srl-vjunosevolved.md](https://github.com/srl-labs/containerlab/blob/6692636103e8c0234a1798e3884f707385568b5c/docs/lab-examples/srl-vjunosevolved.md), and the proper lines in the `mkdocs.yml` file. Also removed `"vr-vjunosevolved", "vr-juniper_vjunosevolved"` entry from [schemas/clab.schema.json](https://github.com/srl-labs/containerlab/blob/6692636103e8c0234a1798e3884f707385568b5c/schemas/clab.schema.json)

steiler commented:
@akielaries would you be willing to also make the ssh keys available in the config?
In the PreDeploy(...) function as part of the params you get the ssh keys, that need to be deployed to the node, to allow for password less access. The SRL and SROS nodes do provision ssh keys to date. Would be nice to have that available for the other kinds as well. I'd be happy to assist.

akielaries commented:
@hellt Sure thing! I went ahead and updated the clab version in the vJunosEvolved lab example doc and removed the vr- prefixes found in references to vJunosEvolved and vJunos-switch. 

@steiler I see how the SROS node is doing this but what entry within the lab's `.yml` definition file would we put to reference SSH keys we want to upload to a given node?

steiler commented:
> @steiler I see how the SROS node is doing this but what entry within the lab's `.yml` definition file would we put to reference SSH keys we want to upload to a given node?

In the pre deploy phase you get all the keys we find... *.pub files, agent registered once etc. In srl /sros we install them all for the relevant user.

hellt commented:
Thanks @akielaries 
The ssh keys you may work on in a consequent PR
